lib/openstax/aws/stack.rb

```diff
@@ -107,6 +176,29 @@ def wait_for_deletion
                            word: "deleted") if !dry_run
     end
 
+    def resource(logical_id)
+      stack_resource = aws_stack.resource(logical_id)
+
+      case stack_resource.resource_type
+      when "AWS::AutoScaling::AutoScalingGroup"
+        name = stack_resource.physical_resource_id
+        client = Aws::AutoScaling::Client.new(region: region)
+        Aws::AutoScaling::AutoScalingGroup.new(name: name, client: client)
+      else
+        raise "'#{stack_resource.resource_type}' is not yet implemented in `Stack#resource`"
+      end
+    end
+
+    def capabilities
+      @capabilities ||= set_capabilities(
+        if OpenStax::Aws.configuration.infer_stack_capabilities
```

Comment:
the dangling ) looks really odd here.  maybe the inner piece should be its own method to make this a bit easier to grok?

Reply:
Yeah this was terrible actually.  It worked but `@x ||= set_x` is also weird.  Refactored to

```ruby
    def capabilities
      set_capabilities(default_capabilities) if @capabilities.nil?
      @capabilities
    end

    def default_capabilities
      if OpenStax::Aws.configuration.infer_stack_capabilities
        template.required_capabilities
      else
        []
      end
    end
```